programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -1880,6 +1771,71 @@ declare_syscall!(
     }
 );
 
+declare_syscallhash!(
+    // Generic Hashing Syscall
+    SyscallHash,
+    fn inner_call<H: HasherImpl>(
+        invoke_context: &mut InvokeContext,
+        vals_addr: u64,
+        vals_len: u64,
+        result_addr: u64,
+        _arg4: u64,
+        _arg5: u64,
+        memory_mapping: &mut MemoryMapping,
+    ) -> Result<u64, Error> {
+        let compute_budget = invoke_context.get_compute_budget();
+        if compute_budget.sha256_max_slices < vals_len {
+            ic_msg!(
+                invoke_context,
+                "Hashing {} sequences in one syscall is over the limit {}",
```

Comment:
this omits the digest name from the log message.  add a `const NAME: &'static str` associated constant to the trait and set it for each